### PR TITLE
Fix SCRIPT_MAX_ARRAYSIZE

### PR DESCRIPTION
**Pull request by AnnieRuru**
[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style
### Pull Request Prelude
- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

## Issue no.1
SCRIPT_MAX_ARRAYSIZE is UINT_MAX -1 ...
so let's try it out
```
prontera,161,185,5	script	test	1_F_MARIA,{
	.@test[2147483648] = 1;
	end;
}
```
oops ~ already got an error
```
[Warning]: script error in file 'npc/zzz.txt' line 2 column 9
    parse_simpleexpr: overflow detected, capping value to INT_MAX
     1: {
*    2:         .@test[2147483648] = 1;
        ~~~~~~~~~~~~~~~^
     3:         end;
     4: }
```
and no need to say that it cannot be a negative value
```
[Warning]: script:getelementofarray: index out of range (-1)
[Debug]: Data: variable name='.@test' index=0
[Debug]: Source (NPC): test9 at prontera (161,185)
```
this means ... the element in array can only hold between 0~2147483647

## Issue No. 2
let's try another thing
```
prontera,161,185,5	script	test9	1_F_MARIA,{
	.@test[0x7FFFFFFF] = 1;
	dispbottom getarraysize(.@test) +"";
	end;
}
```
return ... WHAT ?? `2147483648` ?? shouldn't this throw error ?
```
prontera,161,185,5	script	test9	1_F_MARIA,{
	.@test[0x7FFFFFFF] = 1;
	dispbottom ( getarraysize(.@test) +0 )+"";
	end;
}
```
now this is better, OVERFLOW !! return `-2147483648`

in other words, the maximum array limit should be INT_MAX -1,
because when needed to display in `*getarraysize`, it shouldn't hit the INT limit

### Changes Proposed
Fix `SCRIPT_MAX_ARRAYSIZE`

### Affected Branches
* Master

### Known Issues and TODO List
```
prontera,161,185,5	script	test9	1_F_MARIA,{
	.@test[0x7FFFFFFF] = 1;
	dispbottom ( getarraysize(.@test) +100 )+"";
	end;
}
```
try this out !

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2093)

File: src/map/script.c [hunk 1]
```diff
@@ -16737,7 +16737,7 @@ BUILDIN(explode)
 	temp = aMalloc(len + 1);
 
 	for (i = 0; str[i] != '\0'; i++) {
-		if (str[i] == delimiter && (int64)start + k < (int64)(SCRIPT_MAX_ARRAYSIZE-1)) { // FIXME[Haru]: SCRIPT_MAX_ARRAYSIZE should really be unsigned (and INT32_MAX)
+		if (str[i] == delimiter && start + k < SCRIPT_MAX_ARRAYSIZE) {
```
Comment (4144) [inline src/map/script.c]: why you removed int64? Here probably possible overflow or underflow in ints
in other words what happened if here ``INT_MAX + INT_MAX < INT_MAX - 1``

Comment (AnnieRuru) [reply src/map/script.c]: just initialize 100k element array already let the script engine stuck for 1 minute ....
hahaha fix something that is never meant to happen anyways :laughing:
